Volume/Supporting/Notifications.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Notifications.swift
+//  Volume
+//
+//  Created by Hanzheng Li on 11/17/21.
+//  Copyright © 2021 Cornell AppDev. All rights reserved.
+//
+
+import AppDevAnalytics
+import Foundation
+import UserNotifications
+
+class Notifications: NSObject {
+    static let shared = Notifications()
+    let center = UNUserNotificationCenter.current()
+    
+    private override init() {
+        super.init()
+        center.delegate = self
+    }
+    
+    func requestAuthorization() {
+        center.getNotificationSettings { settings in
+            if settings.authorizationStatus == .notDetermined {
+                self.center.requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+                    if granted {
+                        let event = VolumeEvent.enableNotification.toEvent(.notification, value: "", navigationSource: .unspecified)
+                        AppDevAnalytics.log(event)
+                    } else if let error = error {
+                        print("Error requesting push notification permissions: \(error)")
+                    }
+                }
+            }
+        }
+    }
+    
+    func registerForRemoteNotifications() {
+        if !UIApplication.shared.isRegisteredForRemoteNotifications {
+            DispatchQueue.main.async {
+                UIApplication.shared.registerForRemoteNotifications()
+            }
+        }
+    }
+    
+    func handlePushNotification(userInfo: [AnyHashable: Any]) {
+        if let data = userInfo["data"] as? [String: Any],
+            let notificationType = data["notificationType"] as? String {
+            switch notificationType {
+            case NotificationType.newArticle.rawValue:
+                guard let articleID = data["articleID"] as? String else {
+                    print("Error: missing articleID in new article notification payload")
+                    return
+                }
+                openArticle(id: articleID)
+            default:
+                // later: add case for weekly debrief
+                print("Error: unknown notificationType: \(notificationType)")
+                break
+            }
+            return
```

Comment:
What's the point if this return?

Reply:
To prevent printing the error message on line 62. However, on second thought I think a `guard let` would handle this better.